src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.cs

```diff
@@ -23,65 +25,239 @@ namespace System.Text.RegularExpressions.Generator
+                writer.WriteLine($"    using System.Text.RegularExpressions;");
+                writer.WriteLine($"    using System.Threading;");
+                writer.WriteLine($"");
+                if (compilationDataAndResults.Right.AllowUnsafe)
+                {
+                    writer.WriteLine($"    [SkipLocalsInit]");
+                }
+                writer.WriteLine($"    [{s_generatedCodeAttribute}]");
+                writer.WriteLine($"    [EditorBrowsable(EditorBrowsableState.Never)]");
+                writer.WriteLine($"    internal static class {generatedClassName}");
+                writer.WriteLine($"    {{");
+
+                // Emit each Regex-derived type.
+                writer.Indent += 2;
+                foreach (object? result in results)
+                {
+                    if (result is ValueTuple<RegexType, string, Diagnostic> limitedSupportResult)
+                    {
+                        EmitRegexLimitedBoilerplate(writer, limitedSupportResult.Item1.Method, limitedSupportResult.Item1.GeneratedId, limitedSupportResult.Item2);
+                    }
+                    else if (result is ValueTuple<RegexType, string, Dictionary<string, string[]>> regexImpl)
+                    {
+                        EmitRegexDerivedImplementation(writer, regexImpl.Item1.Method, regexImpl.Item1.GeneratedId, regexImpl.Item2);
+                    }
+                    writer.WriteLine();
+                }
+                writer.Indent -= 2;
+
+                // If any of the Regex-derived types asked for helper methods, emit those now.
+                if (requiredHelpers.Count != 0)
+                {
+                    writer.Indent += 2;
+                    writer.WriteLine($"private static class {HelpersTypeName}");
+                    writer.WriteLine($"{{");
+                    writer.Indent++;
+                    foreach (KeyValuePair<string, string[]> helper in requiredHelpers)
+                    {
+                        foreach (string value in helper.Value)
+                        {
+                            writer.WriteLine(value);
+                        }
+                        writer.WriteLine();
                     }
+                    writer.Indent--;
+                    writer.WriteLine($"}}");
+                    writer.Indent -= 2;
                 }
 
-                context.AddSource("RegexGenerator.g.cs", string.Join(Environment.NewLine, code));
+                writer.WriteLine($"    }}");
+                writer.WriteLine($"}}");
+
+                // Save out the source
+                context.AddSource("RegexGenerator.g.cs", sw.ToString());
             });
         }
+
+        /// <summary>Computes a hash of the string.</summary>
+        /// <remarks>
+        /// Currently an FNV-1a hash function. The actual algorithm used doesn't matter; just something
+        /// simple to create a deterministic, pseudo-random value that's based on input text.
```

Comment:
Might be a dumb question, but why do we need to base it on the Assembly name as opposed of just creating a completely random one?

Reply:
We need the generated code to be deterministic for reproducible builds... same inputs, same outputs.